apps/state/lib/state/stops_on_route.ex

```diff
@@ -125,12 +125,43 @@ defmodule State.StopsOnRoute do
   defp do_gather_route_direction(route, direction_id, trips) do
     global_stop_id_order = order_stop_ids_for_trips(route, direction_id, trips)
 
-    trips
-    |> Enum.group_by(fn trip ->
-      ignore? = ignore_trip_for_route?(trip)
-      {ignore?, trip.shape_id, trip.service_id, trip.direction_id}
-    end)
-    |> Enum.flat_map(&do_gather_direction_group(route, global_stop_id_order, &1))
+    # stops broken down by shape
+    shape_records =
+      trips
+      |> Enum.group_by(fn trip ->
+        ignore? = ignore_trip_for_route?(trip)
+        {ignore?, trip.shape_id, trip.service_id, trip.direction_id}
+      end)
+      |> Enum.flat_map(&do_gather_direction_group(route, global_stop_id_order, &1))
+
+    # stops not broken down by shape or pattern
+    other_records =
+      trips
+      |> Enum.group_by(fn trip ->
+        {ignore_trip_route_pattern?(trip), :all, trip.service_id, trip.direction_id}
+      end)
+      |> Enum.flat_map(&do_gather_direction_group(route, global_stop_id_order, &1))
+
+    Enum.concat(shape_records, other_records)
+  end
+
+  defp do_gather_direction_group(
```

Comment:
This looks pretty much identical to the [existing function](https://github.com/mbta/api/blob/340dbcdba656c33c91e3d0c26c7a8f6744074f52/apps/state/lib/state/stops_on_route.ex#L167-L182)? The only difference seems to be that new function uses `:all` atom as shape id, but I don't see why we cannot just pass it as the argument to the existing function. Or am I missing something?

Reply:
This version uses `ignore_trip_route_pattern?` rather than `ignore_trip_for_route?`. The new function uses the route patterns to decide which trips to use, rather than the shapes.

Reply:
I understand that new version of `do_gather_route_direction` is different from the old one; what I'm talking about is that you now have two almost identical versions of `do_gather_direction_group` -- this one, and the one on lines 167-182. These functions don't use `ignore_trip_route_pattern?` or `ignore_trip_for_route?` internally.

Reply:
Ah, I see; thanks!

Reply:
fixed!